Add a workspaces section to our to-level package.json

markbiek opened the pull request:
### Changes proposed in this Pull Request:

This sets it up so that running `npm i` in the project root will also insall the dependencies defined by `package.json` in the `./scripts/` directory.

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Related to #1096

<!-- The next section is mandatory. If your PR doesn't require testing, please indicate that you are purposefully omitting instructions. -->

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Otherwise, please include detailed instructions on how these changes can be tested. Please, make sure to review and follow the guide for writing high-quality testing instructions below. -->

- [ ] Have you followed the [Writing high-quality testing instructions guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions)?

1. Run the command `rm -rf ./node_modules ./scripts/node_modules` in the project root.
2. Run the command `npm i` in the project root.
3. Verify that there is a `node_modules/` directory in both the project root _and_ the `scripts/` directory.

<!-- End testing instructions -->

### Other information:

-   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?

<!-- Mark completed items with an [x] -->

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

markbiek committed:
Add a workspaces section to our to-level package.json

This sets it up so that running `npm i` in the project root will also
insall the dependencies defined by `package.json` in the `./scripts/`
directory.

andres-blanco reviewed (changes requested):
After running `npm i` I have a change to `package-json.lock` in my working copy:
<img width="777" alt="image" src="https://user-images.githubusercontent.com/375980/235953193-c06e0c9d-d86a-47fc-9e0e-a62061d300f9.png">

This is the diff:
```diff
diff --git a/package-lock.json b/package-lock.json
index 69c3989..5c2ba62 100644
--- a/package-lock.json
+++ b/package-lock.json
@@ -25484,6 +25484,7 @@
                        }
                },
                "scripts": {
+                       "name": "release-automation",
                        "version": "1.0.0",
                        "dependencies": {
                                "chalk": "^5.2.0",
```

markbiek committed:
Package lock update

markbiek commented:
@andres-blanco I think that's from one of the dependencies we introduced in an earlier commit. I've rebased and committed the updated `package-lock.json` and it should be fine now.

Removing both `node_modules/` and `scripts/node_modules/` and running `npm i` shouldn't change the `package-lock.json` now.

andres-blanco reviewed (approved):
LGTM!
